[Swift in WebKit] Address some naming conflicts and add some missing imports

rr-codes opened the pull request:
#### 4b3e6b50064d16303a1b09e2280ac49c65adedcc
<pre>
[Swift in WebKit] Address some naming conflicts and add some missing imports
<a href="https://bugs.webkit.org/show_bug.cgi?id=296500">https://bugs.webkit.org/show_bug.cgi?id=296500</a>
<a href="https://rdar.apple.com/156741496">rdar://156741496</a>

Reviewed by Alex Christensen.

Further support being able to use Swift-Cxx interop in WebKit. See details below.

* Source/JavaScriptCore/inspector/InspectorAgentRegistry.h:
* Source/JavaScriptCore/inspector/InspectorTarget.h:
* Source/WTF/wtf/HashSet.h:
* Source/WebCore/PAL/pal/HysteresisActivity.h:
* Source/WebCore/PAL/pal/system/SleepDisabler.h:
* Source/WebCore/platform/KeypressCommand.h:
* Source/WebCore/platform/LayoutUnit.h:
* Source/WebCore/platform/graphics/Color.h:
* Source/WebCore/platform/graphics/IntSize.h:
* Source/WebKit/Platform/IPC/Connection.h:
* Source/WebKit/Platform/IPC/ConnectionHandle.h:
* Source/WebKit/Shared/API/APIString.h:
* Source/WebKit/Shared/ApplePay/WebPaymentCoordinatorProxy.h:
* Source/WebKit/Shared/EditingRange.h:
* Source/WebKit/Shared/NativeWebGestureEvent.h:

Explicitly add missing imports that were previously already transitively imported.
This is required for Swift to be able to comprehend these files.

* Source/WebKit/Scripts/generate-serializers.py:
(generate_header):
* Source/WebKit/Shared/cf/CFTypes.serialization.in:

The `WebPage` Swift API has a public API property of type `SecTrust`. However, in
the generated serializers header file, there is a forward declaration of `SecTrust`
itself. Therefore, this causes a name conflict because the public API property tries
to use the forward declaration version, which fails because it itself is not public.
This is fixed by importing the relevant header directly and removing the forward declaration
so that there is no conflict.

* Source/WebKit/UIProcess/Automation/BidiBrowsingContextAgent.cpp:
(WebKit::BidiBrowsingContextAgent::getBrowsingContextID const):
(WebKit::BidiBrowsingContextAgent::getNavigableInfo):
(WebKit::BidiBrowsingContextAgent::getNextTree):
(WebKit::BidiBrowsingContextAgent::getTree):
* Source/WebKit/UIProcess/Automation/BidiScriptAgent.cpp:
(WebKit::BidiScriptAgent::callFunction):
(WebKit::BidiScriptAgent::evaluate):
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::getNextContext):
(WebKit::WebAutomationSession::getBrowsingContexts):
(WebKit::WebAutomationSession::getBrowsingContext):
(WebKit::WebAutomationSession::createBrowsingContext):
(WebKit::WebAutomationSession::computeElementLayout):
(WebKit::WebAutomationSession::getAllCookies):
(WebKit::WebAutomationSession::performMouseInteraction):

When `Protocol` is not expressed as `Inspector::Protocol`, it conflicts with the Objective C and Swift
built-in `Protocol` type. Fix this by prefixing all uses with `Inspector::`, which is beneficial anyways
since it is what the majority of the uses already do.

Canonical link: <a href="https://commits.webkit.org/297886@main">https://commits.webkit.org/297886@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dae3af264e14dc00d9183e29a5a42a9f3183cfca

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113279 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33019 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23492 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119487 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/63955 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ❌ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115241 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33667 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41577 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86213 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/63955 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116226 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26893 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101900 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66539 "Passed tests") | | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/112711 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26158 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20027 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63241 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/105796 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96267 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20101 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122704 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/111895 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40357 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/30107 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95059 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40742 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98113 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94805 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24188 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39954 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17772 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/36493 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40243 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45742 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/136125 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39884 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36527 "Found 2 new JSC stress test failures: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention-2.js.default, wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention.js.default (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43217 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41621 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Scripts/generate-serializers.py:
```diff
@@ -679,6 +679,10 @@ def generate_header(serialized_types, serialized_enums, additional_forward_decla
         result.append(f'#include {header}')
 
+    result.append(f'#if USE(CF)')
+    result.append(f'#include <Security/SecTrust.h>')
```
Please run make -C Source/WebKit/Scripts/webkit/tests when changing this.
Also, this is suspicious.  It looks like it'll include a header from a widely-included header.  Can this be avoided?

rr-codes committed:
[Swift in WebKit] Address some naming conflicts and add some missing imports
https://bugs.webkit.org/show_bug.cgi?id=296500
rdar://156741496

Reviewed by Alex Christensen.

Further support being able to use Swift-Cxx interop in WebKit. See details below.

* Source/JavaScriptCore/inspector/InspectorAgentRegistry.h:
* Source/JavaScriptCore/inspector/InspectorTarget.h:
* Source/WTF/wtf/HashSet.h:
* Source/WebCore/PAL/pal/HysteresisActivity.h:
* Source/WebCore/PAL/pal/system/SleepDisabler.h:
* Source/WebCore/platform/KeypressCommand.h:
* Source/WebCore/platform/LayoutUnit.h:
* Source/WebCore/platform/graphics/Color.h:
* Source/WebCore/platform/graphics/IntSize.h:
* Source/WebKit/Platform/IPC/Connection.h:
* Source/WebKit/Platform/IPC/ConnectionHandle.h:
* Source/WebKit/Shared/API/APIString.h:
* Source/WebKit/Shared/ApplePay/WebPaymentCoordinatorProxy.h:
* Source/WebKit/Shared/EditingRange.h:
* Source/WebKit/Shared/NativeWebGestureEvent.h:

Explicitly add missing imports that were previously already transitively imported.
This is required for Swift to be able to comprehend these files.

* Source/WebKit/Scripts/generate-serializers.py:
(generate_header):
* Source/WebKit/Shared/cf/CFTypes.serialization.in:

The `WebPage` Swift API has a public API property of type `SecTrust`. However, in
the generated serializers header file, there is a forward declaration of `SecTrust`
itself. Therefore, this causes a name conflict because the public API property tries
to use the forward declaration version, which fails because it itself is not public.
This is fixed by importing the relevant header directly and removing the forward declaration
so that there is no conflict.

* Source/WebKit/UIProcess/Automation/BidiBrowsingContextAgent.cpp:
(WebKit::BidiBrowsingContextAgent::getBrowsingContextID const):
(WebKit::BidiBrowsingContextAgent::getNavigableInfo):
(WebKit::BidiBrowsingContextAgent::getNextTree):
(WebKit::BidiBrowsingContextAgent::getTree):
* Source/WebKit/UIProcess/Automation/BidiScriptAgent.cpp:
(WebKit::BidiScriptAgent::callFunction):
(WebKit::BidiScriptAgent::evaluate):
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::getNextContext):
(WebKit::WebAutomationSession::getBrowsingContexts):
(WebKit::WebAutomationSession::getBrowsingContext):
(WebKit::WebAutomationSession::createBrowsingContext):
(WebKit::WebAutomationSession::computeElementLayout):
(WebKit::WebAutomationSession::getAllCookies):
(WebKit::WebAutomationSession::performMouseInteraction):

When `Protocol` is not expressed as `Inspector::Protocol`, it conflicts with the Objective C and Swift
built-in `Protocol` type. Fix this by prefixing all uses with `Inspector::`, which is beneficial anyways
since it is what the majority of the uses already do.

Canonical link: https://commits.webkit.org/297886@main
